src/zenml/stack_stores/rest_stack_store.py

```diff
@@ -0,0 +1,359 @@
+            self.get(f"/stacks/get-configurations/{name}")
+        )
+
+    @property
+    def stack_configurations(self) -> Dict[str, Dict[StackComponentType, str]]:
+        """Configurations for all stacks registered in this stack store.
+
+        Returns:
+            Dictionary mapping stack names to Dict[StackComponentType, str]'s
+        """
+        body = self.get("/stacks/get-configurations")
+        if not isinstance(body, dict):
+            raise ValueError(
+                f"Bad API Response. Expected dict, got {type(body)}"
+            )
+        return {
+            key: self._parse_stack_configuration(value)
+            for key, value in body.items()
+        }
+
+    def register_stack_component(
+        self,
+        component: StackComponentWrapper,
+    ) -> None:
+        """Register a stack component.
+
+        Args:
+            component: The component to register.
+
+        Raises:
+            StackComponentExistsError: If a stack component with the same type
+                and name already exists.
+        """
+        self.post("/components/register", body=component)
+
+    def deregister_stack(self, name: str) -> None:
+        """Delete a stack from storage.
+
+        Args:
+            name: The name of the stack to be deleted.
+
+        Raises:
+            KeyError: If no stack exists for the given name.
+        """
+        self.get(f"/stacks/delete/{name}")
+
+    # Custom implementations:
+
+    @property
+    def stacks(self) -> List[StackWrapper]:
+        """All stacks registered in this repository."""
+        body = self.get("/stacks/get")
+        if not isinstance(body, list):
+            raise ValueError(
+                f"Bad API Response. Expected list, got {type(body)}"
+            )
+        return [StackWrapper.parse_obj(s) for s in body]
+
+    def get_stack(self, name: str) -> StackWrapper:
+        """Fetch a stack by name.
+
+        Args:
+            name: The name of the stack to retrieve.
+
+        Returns:
+            StackWrapper instance if the stack exists.
+
+        Raises:
+            KeyError: If no stack exists for the given name.
+        """
+        return StackWrapper.parse_obj(self.get(f"/stacks/get/{name}"))
+
+    def register_stack(self, stack: StackWrapper) -> Dict[str, str]:
+        """Register a stack and its components.
+
+        If any of the stacks' components aren't registered in the stack store
+        yet, this method will try to register them as well.
+
+        Args:
+            stack: The stack to register.
+
+        Returns:
+            metadata dict for telemetry or logging.
+
+        Raises:
+            StackExistsError: If a stack with the same name already exists.
+            StackComponentExistsError: If a component of the stack wasn't
+                registered and a different component with the same name
+                already exists.
+        """
+        try:
+            body = self.post("/stacks/register", stack)
```

Comment:
Does it make sense to have all these endpoint strings as constants somewhere (they're used in both the rest stack store as well as the actual server code)?